pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -456,6 +458,75 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, err
 	}
 
+	if cd.Spec.MachineManagementStrategy != nil && cd.Spec.MachineManagementStrategy.Strategy == "Central" {
```

Comment:
Do you think we could move this to a separate controller? In general this makes the code easier to organize and test, but let me know if you feel it's not a good fit.